File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteNode(self, node):
        """
        :type node: ListNode
        :rtype: void Do not return anything, modify node in-place instead.
        """
        cur = node
        while cur.next.next != None:
            cur.val = cur.next.val
            cur = cur.next
        
        cur.val = cur.next.val
        cur.next = None
        
        

File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

/*
func deleteNode(node *ListNode) {
    current := node
    for current.Next.Next != nil {
        current.Val = current.Next.Val
        current = current.Next
    }
    current.Val = current.Next.Val
    current.Next = nil
}

func deleteNode(node *ListNode) {
    node.Val = node.Next.Val
    node.Next = node.Next.Next
}
*/

func deleteNode(node *ListNode) {
    *node = *node.Next
}